recipe-server/client/control/state/app/experimenter/actions.js

```diff
@@ -0,0 +1,111 @@
+import { createRecipe } from 'control/state/app/recipes/actions';
+import {
+  EXPERIMENTS_RECEIVE,
+  EXPERIMENT_ACCEPTED,
+  EXPERIMENT_REJECTED,
+  EXPERIMENT_SELECT_REJECTED,
+} from 'control/state/action-types';
+
+
+function callExperimenter(url, data) {
+  const options = {
+    method: 'PATCH',
+    headers: new Headers({
+      'Content-Type': 'application/json',
+    }),
+    body: JSON.stringify(data),
+  };
+  return fetch(url, options);
+}
+
+export function getPendingExperimentsUrl(experimenterAPIUrl) {
+  return `${experimenterAPIUrl}/experiments/?status=Pending`;
+}
+
+export function escapeClientMatching(experiment) {
+  // TODO JEXL has no ability to add comment strings so we have to
+  // massage the incoming client matching string into something it
+  // will accept
+  const cleanedMatching = experiment.client_matching.replace(/"/gm, "'").replace(/(\r\n|\n|\r)/gm, '" &&\n"');
+  return `"TODO - CONVERT TO JEXL" &&
+"Browser: ${parseFloat(experiment.population_percent)}% of Firefox ${experiment.firefox_version} ${experiment.firefox_channel}" &&
```

Comment:
I'm curious about what a typical string looks like and what you're cleaning out here?

Reply:
We want to communicate the different types of filtering that will go here, but the Experimenter user won't actually formulate it as a JEXL statement, they'll just write it in natural language like 'Please filter for locale en-US only'.  That then needs to be communicated to the Shield/Normandy admin, but because JEXL doesn't support actual comments, the best I could seem to manage was just making it a string literal which the Shield/Normandy admin will then need to convert to a proper JEXL statement.